multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -518,22 +524,34 @@ def transform_label(
             y = self._label_scaler.transform(np.expand_dims(y, axis=-1))[:, 0].astype(np.float32)
         elif self.label_type == ROIS:
             y = y_df  # Do nothing. TODO: Shall we transform this?
-        elif self.label_type == NER_ANNOTATION:
-            # TODO: Add transform_multimodal and process_multimodal.py
+        else:
+            raise NotImplementedError
+
+        return {self._label_column: y}, {self._label_column: self.label_type}
+
+    def transform_multimodal(
+        self,
+        df: pd.DataFrame,
+    ) -> Tuple[Dict[str, NDArray[(Any,), Any]], Dict[str, str]]:
+        ret = {}
+        if self.label_type == NER_ANNOTATION:
+            # text column should always present
             text_column_index = 0  # Currently, we only support one text column.
             x_df = df[self._text_feature_names[text_column_index]]
-            y = self._label_generator.transform(y_df)
             x = self.transform_text(df)[0]
-            # Labelprocessor needs both ner annotations and text.
-            ret = {
-                NER_ANNOTATION: y,
-                TEXT: x[self._text_feature_names[text_column_index]],
-            }
-            return ret, None
+            ret.update(
+                {
+                    TEXT: x[self._text_feature_names[text_column_index]],
```

Comment:
How about directly using `x` instead of using a hardcode `TEXT`?

Reply:
resolved [here](https://github.com/awslabs/autogluon/pull/2220/commits/b66cdb2aa7df0ecce0003a941a965c00a8012ce2)